Sources/CodableKit/Reflection/HiLoDecoder.swift

```diff
@@ -1,44 +1,46 @@
 // MARK: Internal
 
 /// Decodes types as either "hi" or "lo" signal.
-struct HiLoDecoder: Decoder {
+struct HiLoDecoder<Root, Value>: Decoder {
     enum Signal { case hi, lo }
-    
+
     private let ctx: Context
     let codingPath: [CodingKey]
     let userInfo: [CodingUserInfoKey: Any] = [:]
-    
-    init(signal: Signal) {
-        self.init(.init(signal: signal), codingPath: [])
+    let keyPath: KeyPath<Root, Value>?
```

Comment:
hmm I don't see this being used anywhere, am i missing something?

Reply:
I overlooked that one, and you are correct. It is only used to determined the generic types.